lib/morse_translator.rb

```diff
@@ -40,4 +44,31 @@ def initialize
                     " " => " "}
   end
 
+  def translate(string)
+    string_array = string.downcase.chars
+    morse_array = string_array.map do |character|
+      dictionary["#{character}"]
+    end
+    morse_array.join{""}
+  end
+
+
+  def number_of_words(morse_code)
+    morse_code_array = morse_code.split("  ")
+    counter = morse_code_array.length
+    return morse_code_array
+  end
+
+  def morse_to_eng(morse_code)
+    morse_word_array = morse_code.split("  ")
+    num_of_words = morse_word_array.length
+    final = morse_word_array.map do |word|
+      word = word.split(" ")
+      translation = word.map do |code|
+      dictionary.key("#{code}")
+      end
+    end
+    final
```

Comment:
This is very close! Think that you can probably get `final` where it needs to be with something like the following.

```
final.map do |word|
  word.join("")
end.join(" ")
```

Then you could also probably do away with `num_of_words`.